clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -245,13 +246,19 @@ LogicalResult BreakOp::verify() {
 
 void ConditionOp::getSuccessorRegions(
     ArrayRef<Attribute> operands, SmallVectorImpl<RegionSuccessor> &regions) {
-  auto loopOp = cast<LoopOp>(getOperation()->getParentOp());
-
   // TODO(cir): The condition value may be folded to a constant, narrowing
   // down its list of possible successors.
-  // Condition may branch to the body or to the parent op.
-  regions.emplace_back(&loopOp.getBody(), loopOp.getBody().getArguments());
-  regions.emplace_back(loopOp->getResults());
+
+  // Parent is a loop: condition may branch to the body or to the parent op.
+  if (auto loopOp = dyn_cast<LoopOp>(getOperation()->getParentOp())) {
+    regions.emplace_back(&loopOp.getBody(), loopOp.getBody().getArguments());
+    regions.emplace_back(loopOp->getResults());
+  }
+
+  // Parent is an await: condition may branch to resume or suspend regions.
+  auto await = cast<AwaitOp>(getOperation()->getParentOp());
+  regions.emplace_back(&await.getResume(), await.getResume().getArguments());
+  regions.emplace_back(&await.getSuspend(), await.getSuspend().getArguments());
```

Comment:
From the `ready` region, we may branch either to the `suspend` region or to the parent, but never directly from the `ready` to the `resume` region.